packages/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -2655,6 +2655,322 @@ describe("IModelTransformer", () => {
+    const sourceDbPath = IModelTransformerTestUtils.prepareOutputFile(
+      "IModelTransformer",
+      "PreserveIdOnTestModel-Source.bim"
+    );
+    const sourceDb = SnapshotDb.createEmpty(sourceDbPath, {
+      rootSubject: { name: "iModelA" },
+    });
+    Subject.insert(sourceDb, IModel.rootSubjectId, "Subject1");
+    sourceDb.saveChanges();
+
+    const targetDbPath = IModelTransformerTestUtils.prepareOutputFile(
+      "IModelTransformer",
+      "PreserveIdOnTestModel-Target.bim"
+    );
+    const targetDb = SnapshotDb.createEmpty(targetDbPath, {
+      rootSubject: sourceDb.rootSubject,
+    });
+
+    // Execute process() so that elements from source are copied to target
+    const transformer = new IModelTransformer(sourceDb, targetDb, {
+      preserveElementIdsForFiltering: true,
+    });
+    await transformer.process();
+    targetDb.saveChanges();
+
+    let sourceContent = await getAllElementsInvariants(sourceDb);
+    let targetContent = await getAllElementsInvariants(targetDb);
+    expect(targetContent).to.deep.equal(sourceContent);
+
+    // Delete subject 1 from target
+    const code1 = Subject.createCode(
+      targetDb,
+      IModel.rootSubjectId,
+      "Subject1"
+    );
+    const targetSubjectId1 = targetDb.elements.queryElementIdByCode(code1);
+    expect(targetSubjectId1).to.not.be.undefined;
+    targetDb.elements.deleteElement(targetSubjectId1!);
+    targetDb.saveChanges();
+
+    // Calling process() for second time with option to preserve elements in hopes of restoring deleted element
+    const secondTransformer = new IModelTransformer(sourceDb, targetDb, {
+      preserveElementIdsForFiltering: true,
+    });
+    await secondTransformer.process(); // should not throw error: duplicate code (65547) and should re-add deleted element
+    targetDb.saveChanges();
+
+    // verify that deleted element in target is added back - redundant check for explicitness
+    const sourceElementJSON = sourceDb.elements
+      .getElement<Subject>(targetSubjectId1!)
+      ?.toJSON();
+    const deletedElementInTargetJSON = targetDb.elements
+      .getElement<Subject>(targetSubjectId1!)
+      ?.toJSON();
+    expect(sourceElementJSON).to.be.deep.equal(deletedElementInTargetJSON);
+
+    sourceContent = await getAllElementsInvariants(sourceDb);
+    targetContent = await getAllElementsInvariants(targetDb);
+    expect(targetContent).to.deep.equal(sourceContent);
+
+    sourceDb.close();
+    targetDb.close();
+  });
+
+  it("process() with preserveElementIdsForFiltering set to true should update the element properties if element exists with desired id in target", async () => {
+    const sourceDbPath = IModelTransformerTestUtils.prepareOutputFile(
+      "IModelTransformer",
+      "PreserveIdOnTestModel-Source.bim"
+    );
+    const sourceDb = SnapshotDb.createEmpty(sourceDbPath, {
+      rootSubject: { name: "iModelA" },
+    });
+    Subject.insert(sourceDb, IModel.rootSubjectId, "Subject1");
```

Comment:
For this test, it seems useful to save. But for others tests with preserveIdForFiltering set to true, I think it might be useful to query in the target to see if it exists.